### DISCUSSION TITLE
Curso: Fullstack .NET - Aula: Entity Framework - Migrations

### DISCUSSION THREAD

**Discussion by creabahia** [balta-io/forum#575] [Tira Dúvidas] [answered]
Unable to create a 'DbContext' of type ''. The exception 'Object reference not set to an instance of an object.' was thrown while attempting to create an instance. For the different patterns supported at design time, see https://go.microsoft.com/fwlink/?linkid=851728

![image](https://github.com/balta-io/forum/assets/155968020/f0d03dc0-2d12-4a79-9227-3835d9b99052)

Sempre recebo essa mensagem de erro ao tentar adicionar as Migrations.
Se alguém estiver enfrentando o mesmo problema e puder me auxiliar ficaria agradecido.

URL do repositório:
https://github.com/creabahia/Dima.git

**Comment by malighieri** [comment]
Uma das vezes que me deparei com algo assim estava relacionado ao EF não saber qual projeto deveria usar para instanciar o DBContext e rodar a migration. Para esta situação a forma mais simples de resolver é clicar com o botão direito sobre o projeto que tem o DBContext, no seu caso o de API, e clicar na opção **Set As Startup Project** e depois rodar o comando para gerar e incluir a migration. 

Caso isto não resolva e como este problema pode ter múltiplas causas, segue o link de uma situação desta sendo discutida no Stackoverflow com possíveis soluções incluindo esta que passei:
[Unable to create an object of type '[DBContext's Name]'. For the different patterns supported at design time [closed]](https://stackoverflow.com/questions/55123853/unable-to-create-an-object-of-type-dbcontexts-name-for-the-different-patte)

**Comment by creabahia** [comment]
Obrigado por sua resposta @malighieri, mas ainda não encontrei a solução.
Acessei o link de referência e fiz as configurações mencionadas nos tópicos.

**Comment by creabahia** [comment, marked as answer]
Resolvido.
Foi um erro na definição da propriedade HasColumnType("SMALLINT") quando na verdade deveira ser "NVARCHAR".
A mensagem de erro exibida no terminal não é sugestiva mas no fim das contas era esse o problema.
